### PR TITLE
App Submission: lnplus (Lightning Network+)

### PR DESCRIPTION
**Pull request by lnplus**
# App Submission

### App name
Lightning Network+

### 256x256 SVG icon

- https://lnplus.s3.ap-southeast-1.amazonaws.com/umbrel_app_images/icon.svg

### Gallery images

- https://lnplus.s3.ap-southeast-1.amazonaws.com/umbrel_app_images/1.jpg
- https://lnplus.s3.ap-southeast-1.amazonaws.com/umbrel_app_images/2.jpg
- https://lnplus.s3.ap-southeast-1.amazonaws.com/umbrel_app_images/3.jpg

### I have tested my app on:
- [ ] [Umbrel dev environment](https://github.com/getumbrel/umbrel-dev)
- [ ] [Umbrel OS on a Raspberry Pi 4](https://github.com/getumbrel/umbrel-os)
- [x] [Custom Umbrel install on Linux](https://github.com/getumbrel/umbrel#-installation)

### CODE REVIEW

File: lnplus/umbrel-app.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+description: >-
+  Lightning Network Plus (LN+) is a free app that helps lightning node operators to team up and create liquidity swaps. We will walk you through each step to make it an easy process. You will get incoming liquidity, and will be able to grow your channel count and capacity twice as fast, without any extra costs. The app will automatically log you in with your node. You can join a swap or you can create a new one. You can chat with other node operators. LN+ will help you open the channel with a single click, or you can open the channel manually with the app of your choice. You can rate swap participants, and you will receive ratings to build your node's reputation.
+releaseNotes: >-
+  Initial beta version.
+developer: Alan Plus
+website: https://lightningnetwork.plus
+dependencies:
+  - lightning
+repo: https://github.com/Lightning-Network-Plus/lnpclient
+support: https://github.com/Lightning-Network-Plus/lnpclient/issues
```
Comment (aphex3k) [inline lnplus/umbrel-app.yml:16]: This repository doesn't exist and produces a 404 on github.com

**Review by aphex3k — COMMENTED** [review]
I wanted to check out the repo but it either is private or doesn't exist yet...

File: lnplus/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+version: "3.7"
+
+services:
+  app_proxy:
+    environment:
+      APP_HOST: $APP_LNPLUS_IP
+      APP_PORT: $APP_LNPLUS_PORT
+  web:
+    image: lightningnetworkplus/lnpclient:v0.1.10
```
Comment (aphex3k) [inline lnplus/docker-compose.yml:9]: the image is missing the sha256 after the version information.

File: lnplus/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    restart: on-failure
+    stop_grace_period: 1m
+    volumes:
+      - ${APP_LIGHTNING_NODE_DATA_DIR}:/lnd:ro
+    environment:
+      RAILS_ENVIRONMENT: $RAILS_ENVIRONMENT
```
Comment (aphex3k) [inline lnplus/docker-compose.yml:15]: This should read `$APP_LNPLUS_RAILS_ENVIRONMENT` to avoid risk of reading a wrong environment variable.

**Review by 404Nikhil — APPROVED** [review]

File: lnplus/umbrel-app.yml (hunk 1 above)
Comment (lnplus) [reply lnplus/umbrel-app.yml]: Good catch! I made the repo public now. Thank you!

File: lnplus/docker-compose.yml (hunk 2 above)
Comment (lnplus) [reply lnplus/docker-compose.yml]: Thanks! I now added it.

File: lnplus/umbrel-app.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+repo: https://github.com/Lightning-Network-Plus/lnpclient
+support: https://github.com/Lightning-Network-Plus/lnpclient/discussions
+port: 3777
+gallery:
+  - 1.jpg
+  - 2.jpg
+  - 3.jpg
+path: ""
+defaultUsername: ""
+defaultPassword: ""
```
Comment (nevets963) [inline lnplus/umbrel-app.yml:24]: Could you add this submission metadata to your manifest? Here is an example: https://github.com/getumbrel/umbrel-apps/blob/d5750a1053492d4035d7b13d7359689559a96626/bitcoin/umbrel-app.yml#L62-L63

File: lnplus/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+version: "3.7"
+
+services:
+  app_proxy:
+    environment:
+      APP_HOST: $APP_LNPLUS_IP
```
Comment (nevets963) [inline lnplus/docker-compose.yml:6]: As static container IPs are no longer a requirement, we can instead put `lnplus_web_1` here

File: lnplus/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+      APP_PORT: $APP_LNPLUS_PORT
```
Comment (nevets963) [inline lnplus/docker-compose.yml:7]: Change this to the port number 3777

File: lnplus/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  web:
+    image: lightningnetworkplus/lnpclient:v0.1.11@sha256:fab527c4c0d597e78e18a5ece0002d590b311f910249330ab2447b8f36bc76f2
+    restart: on-failure
+    stop_grace_period: 1m
+    volumes:
+      - ${APP_LIGHTNING_NODE_DATA_DIR}:/lnd:ro
+    environment:
+      RAILS_ENVIRONMENT: $APP_LNPLUS_RAILS_ENVIRONMENT
```
Comment (nevets963) [inline lnplus/docker-compose.yml:15]: Unless it's absolutely neccessary for other apps to know your rails environment, we could just hardcode `production` here?

File: lnplus/exports.sh [hunk 8]
```diff
@@ -0,0 +1,3 @@
+export APP_LNPLUS_IP="10.21.21.77"
+export APP_LNPLUS_PORT="3777"
+export APP_LNPLUS_RAILS_ENVIRONMENT="production"
```
Comment (nevets963) [inline lnplus/exports.sh:3]: If you make the changes within the `docker-compose.yml` file, then this `exports.sh` file is no longer neccessary and simplifies your app :)

File: lnplus/umbrel-app.yml [hunk 9]
```diff
@@ -0,0 +1,24 @@
+manifestVersion: 1
+id: lnplus
+category: Lightning Node Management
+name: Lightning Network+
+version: "0.1.11"
+tagline: Get incoming liquidity and double your channels for free
+description: >-
+  Lightning Network Plus (LN+) is a free app that helps lightning node operators to team up and create liquidity swaps. We will walk you through each step to make it an easy process. You will get incoming liquidity, and will be able to grow your channel count and capacity twice as fast, without any extra costs. The app will automatically log you in with your node. You can join a swap or you can create a new one. You can chat with other node operators. LN+ will help you open the channel with a single click, or you can open the channel manually with the app of your choice. You can rate swap participants, and you will receive ratings to build your node's reputation.
```
Comment (nevets963) [inline lnplus/umbrel-app.yml:8]: Perhaps consider some new lines here to head to users more easily read this within the Umbrel UI.

**Review by nevets963 — COMMENTED** [review]
Hey @lnplus Thanks a huge amount for submitting Lightning Network+ into the app store!

I've left some feedback below on the submission that should be quick and easy to resolve.

I've tried to test the app, but I've hit some challenges.

On my `arm64` Umbrel, I am getting a 500 error. The container logs are showing this:
```
/gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/executor.rb:18: `Failed to open TCP connection to localhost:4000 (Address not available - connect(2) for "localhost" port 4000)' (Errno::EADDRNOTAVAIL)
	from /gems/ruby/3.0.0/gems/railties-7.0.3.1/lib/rails/rack/logger.rb:25:in `block in call'
	from /gems/ruby/3.0.0/gems/activesupport-7.0.3.1/lib/active_support/tagged_logging.rb:114:in `block in tagged'
	from /gems/ruby/3.0.0/gems/activesupport-7.0.3.1/lib/active_support/tagged_logging.rb:38:in `tagged'
	from /gems/ruby/3.0.0/gems/activesupport-7.0.3.1/lib/active_support/tagged_logging.rb:114:in `tagged'
	from /gems/ruby/3.0.0/gems/railties-7.0.3.1/lib/rails/rack/logger.rb:25:in `call'
	from /gems/ruby/3.0.0/gems/sprockets-rails-3.4.2/lib/sprockets/rails/quiet_assets.rb:13:in `call'
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/remote_ip.rb:93:in `call'
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/request_id.rb:26:in `call'
	from /gems/ruby/3.0.0/gems/rack-2.2.4/lib/rack/method_override.rb:24:in `call'
	from /gems/ruby/3.0.0/gems/rack-2.2.4/lib/rack/runtime.rb:22:in `call'
	from /gems/ruby/3.0.0/gems/activesupport-7.0.3.1/lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/server_timing.rb:20:in `call'
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/executor.rb:14:in `call'
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/static.rb:23:in `call'
	from /gems/ruby/3.0.0/gems/rack-2.2.4/lib/rack/sendfile.rb:110:in `call'
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/host_authorization.rb:137:in `call'
	from /gems/ruby/3.0.0/gems/railties-7.0.3.1/lib/rails/engine.rb:530:in `call'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/configuration.rb:252:in `call'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/request.rb:77:in `block in handle_request'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/thread_pool.rb:340:in `with_force_shutdown'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/request.rb:76:in `handle_request'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/server.rb:441:in `process_client'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/thread_pool.rb:147:in `block in spawn_thread'
/gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/executor.rb:18:        raise
(rdb:6)
Errno::EADDRNOTAVAIL (Failed to open TCP connection to localhost:4000 (Address not available - connect(2) for "localhost" port 4000)):

app/controllers/application_controller.rb:87:in `get_message_hash'
app/controllers/application_controller.rb:100:in `auth_hash'
app/controllers/application_controller.rb:8:in `get_notification_count'
/gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/debug_exceptions.rb:72: `Failed to open TCP connection to localhost:4000 (Address not available - connect(2) for "localhost" port 4000)' (Errno::EADDRNOTAVAIL)
	from /gems/ruby/3.0.0/gems/railties-7.0.3.1/lib/rails/rack/logger.rb:25:in `block in call'
	from /gems/ruby/3.0.0/gems/activesupport-7.0.3.1/lib/active_support/tagged_logging.rb:114:in `block in tagged'
	from /gems/ruby/3.0.0/gems/activesupport-7.0.3.1/lib/active_support/tagged_logging.rb:38:in `tagged'
	from /gems/ruby/3.0.0/gems/activesupport-7.0.3.1/lib/active_support/tagged_logging.rb:114:in `tagged'
	from /gems/ruby/3.0.0/gems/railties-7.0.3.1/lib/rails/rack/logger.rb:25:in `call'
	from /gems/ruby/3.0.0/gems/sprockets-rails-3.4.2/lib/sprockets/rails/quiet_assets.rb:13:in `call'
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/remote_ip.rb:93:in `call'
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/request_id.rb:26:in `call'
	from /gems/ruby/3.0.0/gems/rack-2.2.4/lib/rack/method_override.rb:24:in `call'
	from /gems/ruby/3.0.0/gems/rack-2.2.4/lib/rack/runtime.rb:22:in `call'
	from /gems/ruby/3.0.0/gems/activesupport-7.0.3.1/lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/server_timing.rb:20:in `call'
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/executor.rb:14:in `call'
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/static.rb:23:in `call'
	from /gems/ruby/3.0.0/gems/rack-2.2.4/lib/rack/sendfile.rb:110:in `call'
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/host_authorization.rb:137:in `call'
	from /gems/ruby/3.0.0/gems/railties-7.0.3.1/lib/rails/engine.rb:530:in `call'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/configuration.rb:252:in `call'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/request.rb:77:in `block in handle_request'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/thread_pool.rb:340:in `with_force_shutdown'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/request.rb:76:in `handle_request'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/server.rb:441:in `process_client'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/thread_pool.rb:147:in `block in spawn_thread'
/gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/debug_exceptions.rb:72:          raise exception
```
It's not obvious to me why it would be attempting to connect to localhost on port 4000 (which in this case mean the container itself on port 4000).

I think tried to test on my `amd64` node, which is a server in the cloud and I am getting a white screen with this error in the logs:
```
/usr/local/lib/ruby/3.0.0/ipaddr.rb:598: `invalid address' (IPAddr::InvalidAddressError)
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/host_authorization.rb:40:in `any?'
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/host_authorization.rb:40:in `allows?'
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/host_authorization.rb:148:in `authorized?'
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/host_authorization.rb:135:in `call'
	from /gems/ruby/3.0.0/gems/railties-7.0.3.1/lib/rails/engine.rb:530:in `call'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/configuration.rb:252:in `call'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/request.rb:77:in `block in handle_request'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/thread_pool.rb:340:in `with_force_shutdown'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/request.rb:76:in `handle_request'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/server.rb:441:in `process_client'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/thread_pool.rb:147:in `block in spawn_thread'
/usr/local/lib/ruby/3.0.0/ipaddr.rb:598:    raise e.class, "#{e.message}: #{addr}"
(rdb:6) /usr/local/lib/ruby/3.0.0/ipaddr.rb:598: `invalid address' (IPAddr::InvalidAddressError)
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/host_authorization.rb:40:in `any?'
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/host_authorization.rb:40:in `allows?'
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/host_authorization.rb:148:in `authorized?'
	from /gems/ruby/3.0.0/gems/actionpack-7.0.3.1/lib/action_dispatch/middleware/host_authorization.rb:135:in `call'
	from /gems/ruby/3.0.0/gems/railties-7.0.3.1/lib/rails/engine.rb:530:in `call'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/configuration.rb:252:in `call'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/request.rb:77:in `block in handle_request'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/thread_pool.rb:340:in `with_force_shutdown'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/request.rb:76:in `handle_request'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/server.rb:441:in `process_client'
	from /gems/ruby/3.0.0/gems/puma-5.6.4/lib/puma/thread_pool.rb:147:in `block in spawn_thread'
/usr/local/lib/ruby/3.0.0/ipaddr.rb:598:    raise e.class, "#{e.message}: #{addr}"
(rdb:6) [ActionDispatch::HostAuthorization::DefaultResponseApp] Blocked host: umbrel.example.com
```
Do you perhaps have it hard-coded to only allow the domain `umbrel.local`?

File: lnplus/docker-compose.yml (hunk 3 above)
Comment (lnplus) [reply lnplus/docker-compose.yml]: OK. Updated. Thank you!

**Comment by lnplus** [comment]
@nevets963 I had to try a bunch of different combinations how to set up the variables until I found what works. Please let me know if you can run it now!

I made all the changes you requested, except for using `lnplus_web_1` for the IP. It appears it would not be accepted this way:
```
networks:
  default:
    ipv4_address: lnplus_web_1
```

Probably I got some formatting wrong. I couldn't find a reference among the other apps, as none had the `networks` part. 

Happy to fix if you wish me to. Any hints of how to make it work would be welcome.

**Comment by nevets963** [comment]
@lnplus This `networks` yaml block can be dropped. The hostname `lnplus_web_1`is something automatically setup by Docker DNS and going forward, we'll be relying/using that. This `networks` is only needed for static IPs which we don't need to use anymore.

Would you know why I was getting the errors that I posted above?

**Comment by lnplus** [comment]
@nevets963 OK. I see. I will do so.

Yes, it was my mistake, now corrected. In the app it was hardcoded to pull data from local at port: 4000, which I used for local development. I had to change it to pull from the lnplus production API. (related: https://lightningnetwork.plus/api_documentation)

**Comment by nevets963** [comment]
@lnplus Ok great! I managed to get it working on my `arm64` Umbrel and I can see available swaps :)

My `amd64` Umbrel is in the cloud where I use a domain name to access it. I get a `Blocked host` error when I open the app. Is the app locked to only allow `umbrel.local` and access via IPs?

**Comment by lnplus** [comment]
> Is the app locked to only allow `umbrel.local` and access via IPs?

@nevets963 Thanks for finding the bug. No, I"m not locking anything. At least, not intentionally. I'm running a test Umbrel on AWS and it works fine **without a domain**.

I will try to add a domain to it to try and reproduce the error. Are you on https or simply https? If the former, how do you make it work with Umbrel?

**Comment by nevets963** [comment]
@lnplus I'm accessing using `http`, e.g. `http://umbrel.example.com`

**Comment by lnplus** [comment]
> @lnplus I'm accessing using `http`, e.g. `http://umbrel.example.com`

@nevets963 Thank you for finding this bug! I made a fix. It should work with a domain too. Let me know!

**Comment by nevets963** [comment]
@lnplus Awesome! That resolved the issue 😄 

I was doing some further testing and hit a couple of bugs:

When I visit: `/for_me`:
Screenshot 2022-10-25 at 21 28 14


When I visit: `/my_swaps`:
Screenshot 2022-10-25 at 21 25 38

When I visit: `/to_dos`:
Screenshot 2022-10-25 at 21 29 41

**Comment by lnplus** [comment]
@nevets963 Thanks for finding this bug too. Sorry for the trouble! I didn't account for the possibilty that LN+ doesn't have your node in the database yet (https://lightningnetwork.plus/nodes). Perhaps your node is private or brand new without channels.

I will come up with a page to handle such situations and push an update soon...

**Comment by lnplus** [comment]
I updated the LN+ app to handle the situation when the node is not yet found on the gossip network (which is the issue caused the errors above). We require the node to be on the gossip network (because others need to be able to review and use the node parameters to open channels), so the app is not yet functional without that, but at least it gives feedback on what to do (open at least a single channel).

On another note, I noticed I can't set a repo anymore for testing [as described here](https://github.com/getumbrel/umbrel-apps/blob/master/README.md#3-testing-the-app-on-umbrel). What is the new procedure now and should we update the linked page to reflect the new instructions perhaps?

Please let me know what is the next step!

**Comment by nevets963** [comment]
@lnplus Ok nice! I'll give this a test now.

About 60 seconds ago, I updated the docs, and going forward this is the best way to checkout a forked Umbrel App Store repo: 
```
sudo ./scripts/repo checkout https://github.com/<username>/umbrel-apps.git
```

**Comment by lnplus** [comment]
@nevets963 Thanks! `checkout` worked. But my app is giving a 500 error on my Linux cloud test Umbrel. I assume you will receive the same error. I'm going to investigate what is happening... Sorry!

**Comment by nevets963** [comment]
Ok no worries, just ping me again once all clear, so I get a notification to test it myself :)

**Comment by lnplus** [comment]
@nevets963 I updated the app now. It tested OK for me on Linux Cloud. Please kindly check!

(I'm not able to test on a Raspi, still waiting for the hardware to arrive. It's out of stock everywhere.)

File: lnplus/docker-compose.yml [hunk 10]
```diff
@@ -0,0 +1,20 @@
+version: "3.7"
+
+services:
+  app_proxy:
+    environment:
+      APP_HOST: lnplus_web_1
+      APP_PORT: "3777"
+      API_URL: "https://lightningnetwork.plus/api/2/"
```
Comment (nevets963) [inline lnplus/docker-compose.yml:8]: This env. var. is being set for the App Proxy service which doesn't accept `API_URL`. I assume this is supposed to be for your `web` service?

I just tested a few things on `arm64` and `amd64`, and I didn't see any issue, so I assume your `web` service is using a default value for `API_URL`? And as this isn't being used by the App Proxy, this is safe to drop this line. Or would you like to move this to the `web` service?

Comment (lnplus) [reply lnplus/docker-compose.yml]: @nevets963 Thanks for educating me on this. I updated both the app and the umbrel-apps repo.

**Review by nevets963 — APPROVED** [review]
@lnplus Thanks for working with me to perfect this PR!

I tested on `arm64` and `amd64`, and everything appears to be working nicely! 😄 

1 tiny thing, I noticed that `RAILS_ENV` is `development`. Are you sure this is supposed to be `development` and not `production`? I see that it works using `development` though...

This is approved from me 🚀 

@mayankchhabra Would you please be able to help us with the App Store graphics to get this launched.

**Comment by lnplus** [comment]
> 1 tiny thing, I noticed that `RAILS_ENV` is `development`. Are you sure this is supposed to be `development` and not `production`? I see that it works using `development` though...

@nevets963 Thank you! Yes, this is intentional for now, and I don't think it's a problem. It doesn't really matter if it runs in _production_ or _development_, because it runs locally anyway. In a future version, I will work on moving the environment variable to _production_.

@mayankchhabra Let me know if the images I linked from the PR are OK.

**Review by aphex3k — APPROVED** [review]
I like the cooperation going on here. Can't wait to use lnplus from my node directly :D

**Comment by mayankchhabra** [comment]
Excellent work here, everyone! So excited to finally see LN+ go live on Umbrel @lnplus. Great job on packaging it :)

The images you linked look A-OK! We made some slight modifications, mainly using a blurred version of the third image you provided (love that one btw!) as the background for the first two. You can check them out here: https://github.com/getumbrel/umbrel-apps-gallery/pull/15

Let us know if you're happy with the images and we'll push LN+ out!

File: lnplus/umbrel-app.yml [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+support: https://github.com/Lightning-Network-Plus/lnpclient/discussions
+port: 3777
+gallery:
+  - 1.jpg
+  - 2.jpg
+  - 3.jpg
+path: ""
+defaultUsername: ""
+defaultPassword: ""
+submitter: Alan Plus
```
[the remark this answers is not in the document]
Comment (lnplus) [reply lnplus/umbrel-app.yml]: OK! Thanks.

**Comment by lnplus** [comment]
@mayankchhabra Looks good. Thanks for your work. Let's push it out!

**Comment by nevets963** [comment]
Gallery assets were merged in and now lnplus 🚀 🚀 

Great work all round here!

**Comment by lnplus** [comment]
Thank you guys! Really appreciate your work testing and updating, etc. I can see the app in the app store now.

If I wanted to create an update in the future, what is the correct procedure? Do I update the app and then create a new pull request?

**Comment by nevets963** [comment]
Yes exactly, create a branch in your `umbrel-apps` fork, make your changes, and remember to bump the version inside `umbrel-app.yml`, and then submit a PR here 😄

**Pull request merged**